docs/specs/ecosystem-service-credit.md

```diff
@@ -0,0 +1,98 @@
+# Messages
+
+## Credit Issuance, Transfer and Consumption
+
+```go
+// MsgCreateCreditClass creates a class of credits and returns a new CreditClassID
+type MsgCreateCreditClass struct {
+  Curator sdk.AccAddress
+  Name string
+  Issuers []sdk.AccAddress
+}
+
+type []byte CreditClassID
+
+// MsgIssueCredit issues a credit to the Holder with the number of Units provided
+// for the provided credit class, polygon, and start and end dates. A new CreditID
+// is returned. It is illegal to issue a credit where the provided polygon and dates
+// overlaps with those of an existing credit of the same class 
+type MsgIssueCredit struct {
+  CreditClass CreditClassID 
+  Polygon geo.Polygon
+  StartDate time.Time
+  EndDate time.Time
+  Units sdk.Dec
+  Issuer sdk.AccAddress
+  Holder sdk.AccAddress
+}
+
+type []byte CreditID
+
+// MsgSendCredit sends the provided number of units of the credit from the from
+// address to the to address
+type MsgSendCredit struct {
+  Credit CreditID
+  From sdk.AccAddress
+  To sdk.AccAddress
+  Units sdk.Dec
+}
+
+// MsgConsumeCredit consumes the provided number of units of the credit, essentially
+// burning or retiring those units. This operation is used to actually use
+// the credit as an offset. Otherwise, the holder of the credit is simply
+// holding the credit as an asset but has not claimed the offset. Once a
+// credit has been consumed, it can no longer be transferred
+type MsgConsumeCredit struct {
+  Credit CreditID
+  Holder sdk.AccAddress
+  Units sdk.Dec
+}
+```
+
+
+## Credit Exchange
```

Comment:
I think the Exchange is a whole different topic and I would place that elsewhere.

Send and Consume is what we need now
Trying to design a marketplace / orderbook / dex takes a bit more design work.
I did start one simple on-chain orderbook here for example: https://github.com/iov-one/tutorial/
And this is a super rough sketch of behavior: https://github.com/iov-one/tutorial/blob/master/x/orderbook/README.md

Please break this section out to another PR.